omnigibson/object_states/overlaid.py

```diff
@@ -0,0 +1,66 @@
+from omnigibson.object_states.kinematics import KinematicsMixin
+from omnigibson.object_states.object_state_base import BooleanState, RelativeObjectState
+from omnigibson.object_states.touching import Touching
+from omnigibson.utils.constants import PrimType
+import omnigibson.utils.transform_utils as T
+from omnigibson.macros import create_module_macros
+
+from scipy.spatial import ConvexHull, HalfspaceIntersection
+from scipy.spatial.qhull import QhullError
+import numpy as np
+import trimesh
+import itertools
+
+# Create settings for this module
+m = create_module_macros(module_path=__file__)
+
+# Percentage of xy-plane of the object's base aligned bbox that needs to covered by the cloth
+m.OVERLAP_AREA_PERCENTAGE = 0.5
+
+# Subsample cloth particle points to fit a convex hull for efficiency purpose
+m.N_POINTS_CONVEX_HULL = 1000
+
+class Overlaid(KinematicsMixin, RelativeObjectState, BooleanState):
+    @staticmethod
+    def get_dependencies():
+        return KinematicsMixin.get_dependencies() + RelativeObjectState.get_dependencies() + [Touching]
+
+    def _set_value(self, other, new_value, use_ray_casting_method=False):
+        raise NotImplementedError("Overlaid state currently does not support setting.")
+
+    def _get_value(self, other):
```

Comment:
Might be helpful (actually, in general, for other complex object states) providing docstring here describing high-level logic of what determines whether the obj is overlaid or not. What is the HalfSpaceIntersection checking, exactly? (sorry, I'm not too familiar with the trimesh terms)

Reply:
Good call! Added more detailed comments and docstring.